### ISSUE TITLE
Command - the configuration and access to a trust context is simplified

### ISSUE THREAD

**Issue by etorreborre** [build-trust/ockam#6170]
The current configuration of a `TrustContext` is a bit involved:

 - [the `project_path` argument is deprecated](https://github.com/build-trust/ockam/blob/c284481fa0513815b01294bbb57e4ac29e18ec3a/implementations/rust/ockam/ockam_command/src/util/api.rs#L399) but still in use in many tests
 - [from some `TrustContextOpts` we make a `TrustContextConfigBuilder` which eventually builds a `TrustContextConfig`](https://github.com/build-trust/ockam/blob/c284481fa0513815b01294bbb57e4ac29e18ec3a/implementations/rust/ockam/ockam_command/src/util/api.rs#L416)
 - from a `TrustContextConfig` we build a `TrustContext` which contains an optional `AuthorityService`. This makes it awkward to access the `AuthorityService` when defined
 - the `AuthorityService` itself is a bit complicated since it can retrieve a credential from different places
    - this has some interaction with the creation of secure channels since we pass a optional credential name to create a secure channel. Should that name be used when the `AuthorityService` is used with a `CredentialRetriever::FromPath`?

Long story short, the purpose of a `TrustContext` is to retrieve credentials for a given identity and there should be a clearer path to how and when this credential is retrieved.

**Comment by mrinalwadhwa** [comment]
related to - #5769

**Comment by nazmulidris** [comment]
@etorreborre Here are some other issues related to this
- https://github.com/build-trust/ockam/issues/6126
- https://github.com/build-trust/ockam/issues/4868
